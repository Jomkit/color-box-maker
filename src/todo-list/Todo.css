button {
    background: none;
    border: none;
    cursor: pointer;
}

.Todo-completeBtn {
    text-decoration: underline;
    color: hsl(0, 0%, 45%);
}

.Todo-completeBtn:hover {
    color: hsl(0, 0%, 28%);

}

.Todo-editBtn {
    text-decoration: underline;
    margin: 5px;
    padding: 3px;
}

.Todo-editBtn:hover {
    background: lightblue;
    border-radius: 3px;
}

.Todo-deleteBtn {
    color: red;
}
