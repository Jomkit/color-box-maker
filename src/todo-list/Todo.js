import React, { useState } from 'react'
import "./Todo.css";
import Form from './Form';

const Todo = ({ id, task, completed, deleteTodo, editTodo, completeTodo }) => {
    const [formOpen, setFormOpen] = useState(false);
    const toggleEdit = () => {setFormOpen(!formOpen)}
  return (
    <>
        <li style={{textDecoration: completed && "line-through"}}>
            Task: {task} 
            <button className='Todo-completeBtn' onClick={completeTodo}>Mark as completed</button>
            <button className='Todo-editBtn' onClick={toggleEdit} >Edit</button>
            <button className='Todo-deleteBtn' onClick={deleteTodo}><span>&#9746;</span></button>
        </li>
        {formOpen && <Form id={id} toggleEdit={toggleEdit} editTodo={editTodo} />}
    </>
  )
}

export default Todo