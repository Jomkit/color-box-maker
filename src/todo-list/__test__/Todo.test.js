import { render, fireEvent } from "@testing-library/react"
import Todo from "../Todo"

it('should render without crashing', () => { 
    render(<Todo />);
})

it('should match snapshot', () => { 
    const { asFragment } = render(<Todo task="Test Task" />);
    expect(asFragment()).toMatchSnapshot();
})

it('should call delete function when delete button is clicked', () => { 
    const mockDelete = jest.fn();

    const { container, queryByText } = render(<Todo task="Test Task" deleteTodo={mockDelete}/>);
    const deleteBtn = container.querySelector(".Todo-deleteBtn");
    
    fireEvent.click(deleteBtn);
    expect(mockDelete).toHaveBeenCalled();
 })

it('should toggle edit form when edit button is clicked', () => { 
    const mockEdit = jest.fn();
    const { container, queryByText } = render(<Todo task="Test Task" editTodo={mockEdit}/>);
    const editBtn = queryByText("Edit");
    fireEvent.click(editBtn);
    expect(container).toContainHTML('</form>');
 })

it('should call completeTodo function when "Mark as completed" button pressed', () => { 
    const mockComplete = jest.fn();
    const { queryByText } = render(<Todo task={"Test task"} completeTodo={mockComplete} />);
    const completeBtn = queryByText("Mark as completed");
    fireEvent.click(completeBtn);

    expect(mockComplete).toHaveBeenCalled();
 })