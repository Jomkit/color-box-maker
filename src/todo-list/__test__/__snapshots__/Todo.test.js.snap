// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match snapshot 1`] = `
<DocumentFragment>
  <li>
    Task: Test Task
    <button
      class="Todo-completeBtn"
    >
      Mark as completed
    </button>
    <button
      class="Todo-editBtn"
    >
      Edit
    </button>
    <button
      class="Todo-deleteBtn"
    >
      <span>
        â˜’
      </span>
    </button>
  </li>
</DocumentFragment>
`;
